cpp/mg_utility/mg_graph.hpp

```diff
@@ -0,0 +1,264 @@
+/// @file graph.hpp
+/// This file contains the graph definitions.
+
+#pragma once
+
+#include <algorithm>
+#include <map>
+#include <unordered_map>
+
+#include "data_structures/graph_data.hpp"
+#include "data_structures/graph_view.hpp"
+#include "mg_exceptions.hpp"
+
+namespace mg_graph {
+
+/// Graph representation.
+template <typename TSize> class Graph : public GraphView<TSize> {
+
+  using TNode = Node<TSize>;
+  using TEdge = Edge<TSize>;
+  using TNeighbour = Neighbour<TSize>;
+
+  static_assert(
+      std::is_same<TSize, uint8_t>::value ||
+          std::is_same<TSize, uint16_t>::value ||
+          std::is_same<TSize, uint32_t>::value ||
+          std::is_same<TSize, uint64_t>::value,
```

Comment:
https://en.cppreference.com/w/cpp/types/is_unsigned
```suggestion
      std::is_unsigned_v<TSize>,
```